ios/test_app.rb

```diff
@@ -68,11 +68,13 @@ def react_native_path(project_root, target_platform)
   react_native_path = platform_config('reactNativePath', project_root, target_platform)
   return Pathname.new(resolve_module(react_native_path)) if react_native_path.is_a? String
 
-  react_native = case target_platform
-                 when :ios then 'react-native'
-                 when :macos then 'react-native-macos'
-                 else raise "Unsupported target platform: #{target_platform}"
-                 end
+  react_native_packages = {
+    'ios' => 'react-native',
+    'macos' => 'react-native-macos',
+    'visionos' => '@callstack/react-native-visionos',
+  }
+  react_native = react_native_packages[target_platform.to_s]
```

Comment:
Use the JSON-style syntax instead:
```suggestion
  react_native_packages = {
    ios: 'react-native',
    macos: 'react-native-macos',
    visionos: '@callstack/react-native-visionos',
  }
  react_native = react_native_packages[target_platform]
```